### DISCUSSION TITLE
Lesson 4: Input Privacy can prevent Copying Problem?

### DISCUSSION THREAD

**Discussion by cupidchan** [OpenMined/courses#126] [Course: Our Privacy Opportunity] [answered]
At around 9:15 of the course video, Andrew mentioned that Copying problem can be prevented by Input Privacy. However, Input Privacy only makes sure that other contributors cannot see each other input and guarantees that no one in the middle can view the information in transfer. That means the receiver gets the information, the sender still cannot control how the information being used downstream by the receiver. Hence the Copying Problem will still exist, right?

**Comment by iamtrask** [comment, marked as answer]
Fantastic question! So the short answer is yes (kindof) - but the long answer is more interesting.

1) the only time when information needs to actually be an "output" is when someone needs to take a real-world action. And the output only actually needs to be the "action to be taken" if the decision making process can be itself put inside the information flow.
2) Output privacy can also reduce the amount that is actually released - and reduce the confidence by which it is known. So while the copy problem might still exist - you may lose 100% confideence that you are copying the exact result.

(1) is perhaps the most surprising and takes some time to really settle in - there's a lot of creative ways to basically never release information when you move all but the "decisions X person should take in the physical world" inside the information flow.

**Reply by cupidchan** [reply]
Thanks @iamtrask! That makes sense. I think that's the reason why we need all 5 building blocks of Structured Transparency to complete the picture.

**Reply by rubenmak** [reply]
@iamtrask @cupidchan I had a similar thoughts around the input and output privacy 'solving the copy problem', also in relation to the privacy auction concept mentioned in lesson 5. I think 'solving the copy problem' should be nuanced a bit.

Basically, you still have a 'copy problem', you only truly solved the bundling problem. If a company pays for the released epsilon to get a specific aggregate, the company can still sell this this aggregate to everybody else. In other words, it doesn't seem the case that the initial owner / person owning the data can get a financial reward every time the data is used. You sell the specific operation of the data once and this includes selling the aggregate to all other companies. The same thing holds for a trained model, which can just be sold to another company that can further iterate on the model using transfer learning. 

I think you kind a only solve the 'copy problem' if a consumer uses a ML model without the ML model training on that consumers data. But still, I'd consider it more of a bundling problem than a copy problem that's being solved. I.e. the information flow is designed to output to the user only, and the company owning the ML model learns isn't in the output.

What do you think? Anything I'm missing here about how the copy problem can be truly solved? Maybe from a different perspective: the bundling problem is being amplified by the copy problem, so solving the bundling problem seriously reduces the severity of the copy problem.

**Comment by cupidchan** [comment]
@rubenmak , I put more thoughts on your comments. I realize that it depends on how to put the "copy problem" in aspect. For example, if you take the raw data, say my age and enforce input privacy so that you only allow the aggregated result as the output. Provided output privacy is also in place, no one can "copy" my age because it's practically hidden. 

However, I do agree with you that there is no way to limit the receiver to propagate the aggregated result (deduced from my age) further downstream. In this sense, "copy" problem still exists.
